components/Proposals/MintConditionalTokenCard.tsx

```diff
@@ -1,24 +1,43 @@
-import { useCallback, useState } from 'react';
-import { Button, Fieldset, Group, Text, TextInput } from '@mantine/core';
+import { useCallback, useEffect, useState } from 'react';
+import { Button, Fieldset, Group, Text, TextInput, SegmentedControl, Center, Box } from '@mantine/core';
 import numeral from 'numeral';
 import Link from 'next/link';
-import { IconExternalLink } from '@tabler/icons-react';
-import { Token } from '@/hooks/useTokens';
+import { IconExternalLink, IconCurrencyDollar, IconLetterM } from '@tabler/icons-react';
+// import { Token } from '@/hooks/useTokens';
 import { useProposal } from '@/contexts/ProposalContext';
 import { useTransactionSender } from '../../hooks/useTransactionSender';
 import { useExplorerConfiguration } from '@/hooks/useExplorerConfiguration';
 import { useBalance } from '../../hooks/useBalance';
 import { NUMERAL_FORMAT } from '../../lib/constants';
+import { useTokens, Token } from '@/hooks/useTokens';
 
-export function MintConditionalTokenCard({ token }: { token: Token }) {
+export function MintConditionalTokenCard() {
   const sender = useTransactionSender();
   const { markets, mintTokensTransactions } = useProposal();
   const { generateExplorerLink } = useExplorerConfiguration();
+  const { tokens } = useTokens();
 
   if (!markets) return null;
 
   const [mintAmount, setMintAmount] = useState<number>();
-  const fromBase = markets.baseVault.underlyingTokenMint.equals(token.publicKey);
+  const [token, setToken] = useState<string>('meta');
+  const [_token, _setToken] = useState<Token>(tokens.meta);
+
+  const updateTokenValue = () => {
+    if (token === 'meta') {
+      _setToken(tokens.meta);
+    }
+    if (token === 'usdc') {
+      _setToken(tokens.usdc);
+    }
+  };
+
+  useEffect(() => {
+    updateTokenValue();
+  }, [token]);
+
+  // TODO: Fetch the two tokens
+  const fromBase = markets.baseVault.underlyingTokenMint.equals(_token.publicKey);
```

Comment:
we could memoize fromFail, fromPass, vaultFail & vaultPass, then set fromBase and vault on _token change. Do we want to do that in that PR ?

Reply:
Anything that reduces overhead is best effort. So moving towards that is critical.